docs/src/main/asciidoc/security-openid-connect-multitenancy.adoc

```diff
@@ -1110,6 +1110,111 @@ The default tenant configuration is automatically disabled when `quarkus.oidc.au
 
 Be aware that tenant-specific configurations can also be disabled, for example: `quarkus.oidc.tenant-a.tenant-enabled=false`.
 
+[[programmatic-startup]]
+== Programmatic OIDC start-up for multitenant application
+
+Static OIDC tenants can be created programmatically like in the example below:
+
+[source,java]
+----
+package io.quarkus.it.oidc;
+
+import io.quarkus.oidc.Oidc;
+import io.quarkus.oidc.OidcTenantConfig;
+import jakarta.enterprise.event.Observes;
+
+public class OidcStartup {
+
+    void observe(@Observes Oidc oidc) { <1>
+        oidc.create(OidcTenantConfig.authServerUrl("http://localhost:8180/realms/tenant-one").tenantId("tenant-one").build());
+        oidc.create(OidcTenantConfig.authServerUrl("http://localhost:8180/realms/tenant-two").tenantId("tenant-two").build());
+    }
+
+}
+----
+<1> Static tenants must be created in a synchronous manner from the observer method.
+
+The code above is a programmatic equivalent to the following configuration in the `application.properties` file:
+
+[source,properties]
+----
+quarkus.oidc.tenant-one.auth-server-url=http://localhost:8180/realms/tenant-one
+quarkus.oidc.tenant-two.auth-server-url=http://localhost:8180/realms/tenant-two
+----
+
+You can also create static tenants based on the tenants configured in the `application.properties` file.
+For example, if you need to add only tenant paths programmatically, you can:
+
+.Static application.properties
+[source,properties]
+----
+quarkus.oidc.tenant-two.auth-server-url=http://localhost:8180/realms/tenant-two
+quarkus.oidc.tenant-two.tenant-paths=/api/tenant-two/*
+----
+
+.Additional property set programmatically
+[source,java]
+----
+package io.quarkus.it.oidc;
+
+import io.quarkus.oidc.Oidc;
+import io.quarkus.oidc.OidcTenantConfig;
+import jakarta.enterprise.event.Observes;
+
+public class OidcStartup {
+
+    void observe(@Observes Oidc oidc, OidcConfig config) {
+        oidc.create(OidcTenantConfig
+                .builder(config.namedTenants().get("tenant-two"))
+                .tenantPaths(getAdditionalTenantPath()) <1>
+                .build()
+        );
+    }
+
+    private String getAdditionalTenantPath() {
+        return "/additional-tenant-path";
+    }
+
+}
+----
+<1> The tenant `tenant-two` will have two paths, `/additional-tenant-path` and `/api/tenant-two/*`.
+
+Dynamic OIDC tenants resolved with the <<annotations-tenant-resolver>> can be created whenever necessary:
```

Comment:
I'm not sure about this one. Annotations can only be used to resolving static tenants AFAIK ?

Reply:
I changed that :-) I have a test for it. It is not important change, I can drop it. But in that case, I don't see a point of keeping `Oidc` as a bean. We could drop it in that case.

Reply:
Hi Michal, can we deal with it later ? I'm not really sure about talking about static and dynamic tenants in scope of the initial PR introducing this feature. It would be definitely a dedicated PR IMHO in any case. 

FYI, it sounds promising, the whole idea of allocating tenants dynamically to match the annotations. The dynamism is limited but it kind of introduces some tenant-name and request path restriction, but please, let's not deal with it in this PR

Reply:
I'll drop registering dynamic tenants from this PR and we can discuss it in a separate PR later. I don't think it is important use case